From: Arnaud Rebillout <arnaudr@kali.org>
Date: Fri, 17 Nov 2023 10:46:21 +0700
Subject: Dual: mirrors, selection: Drop IsHTTPS

The method IsHTTPS is trivial and it's only used in http/selection.go,
in the function Selection(). This function will be refactored in the
next commits, and the method IsHTTPS() won't be needed anymore.

To make the changes easier to follow, let's start by removing the
IsHTTPS() function now.
---
 http/selection.go  | 4 ++--
 mirrors/mirrors.go | 5 -----
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/http/selection.go b/http/selection.go
index 03d0bbb..b60089a 100644
--- a/http/selection.go
+++ b/http/selection.go
@@ -77,11 +77,11 @@ func (h DefaultEngine) Selection(ctx *Context, cache *mirrors.Cache, fileInfo *f
 			}
 			goto discard
 		}
-		if ctx.SecureOption() == WITHTLS && !m.IsHTTPS() {
+		if ctx.SecureOption() == WITHTLS && !strings.HasPrefix(m.HttpURL, "https://") {
 			m.ExcludeReason = "Not HTTPS"
 			goto discard
 		}
-		if ctx.SecureOption() == WITHOUTTLS && m.IsHTTPS() {
+		if ctx.SecureOption() == WITHOUTTLS && strings.HasPrefix(m.HttpURL, "https://") {
 			m.ExcludeReason = "Not HTTP"
 			goto discard
 		}
diff --git a/mirrors/mirrors.go b/mirrors/mirrors.go
index 68d1d37..99f7c22 100644
--- a/mirrors/mirrors.go
+++ b/mirrors/mirrors.go
@@ -95,11 +95,6 @@ func (m *Mirror) Prepare() {
 	m.ExcludedCountryFields = strings.Fields(m.ExcludedCountryCodes)
 }
 
-// IsHTTPS returns true if the mirror has an HTTPS address
-func (m *Mirror) IsHTTPS() bool {
-	return strings.HasPrefix(m.HttpURL, "https://")
-}
-
 // IsUp returns true if the mirror is up
 func (m *Mirror) IsUp() bool {
 	// Up over both HTTP and HTTPS?
